snapshot-parser/src/validator_meta.rs

```diff
@@ -112,7 +106,124 @@ pub fn generate_validator_collection(bank: &Arc<Bank>) -> anyhow::Result<Validat
+        if account.data[0..8] == TIP_DISTRIBUTION_ACCOUNT_DISCRIMINATOR {
+            let epoch_created_at = u64::from_le_bytes(
+                account.data[EPOCH_CREATED_AT_BYTE_INDEX..EPOCH_CREATED_AT_BYTE_INDEX + 8]
+                    .try_into()
+                    .map_err(|e| {
+                        anyhow::anyhow!(
+                            "Failed to parse epoch_created_at for account {}: {:?}",
+                            pubkey,
+                            e
+                        )
+                    })?,
+            );
+            if epoch_created_at == epoch {
+                let mev_commission = u16::from_le_bytes(
+                    account.data[VALIDATOR_COMMISSION_BPS_BYTE_INDEX..VALIDATOR_COMMISSION_BPS_BYTE_INDEX + 2]
+                        .try_into()
+                        .map_err(|e| anyhow::anyhow!("Failed to parse validator_commission_bps (mev commission) for account {}: {:?}", pubkey, e))?,
+                );
+                let vote_account: Pubkey = account.data
+                    [VALIDATOR_VOTE_ACCOUNT_BYTE_INDEX..VALIDATOR_VOTE_ACCOUNT_BYTE_INDEX + 32]
+                    .try_into()
+                    .map_err(|e| {
+                        anyhow::anyhow!(
+                            "Failed to parse vote account for account {}: {:?}",
+                            pubkey,
+                            e
+                        )
+                    })?;
+                jito_mev_metas.push(JitoMevMeta {
+                    vote_account,
+                    mev_commission,
+                });
+            }
+        }
+    }
+
+    Ok(jito_mev_metas)
+}
+
+pub fn generate_validator_collection(bank: &Arc<Bank>) -> anyhow::Result<ValidatorMetaCollection> {
+    assert!(bank.is_frozen());
+
+    let EpochInfo {
+        epoch,
+        absolute_slot,
+        ..
+    } = bank.get_epoch_info();
+
+    let validator_rate = bank
+        .inflation()
+        .validator(bank.slot_in_year_for_inflation());
+    let capitalization = bank.capitalization();
+    let epoch_duration_in_years = bank.epoch_duration_in_years(epoch);
+    let validator_rewards =
+        (validator_rate * capitalization as f64 * epoch_duration_in_years) as u64;
+
+    let vote_account_metas = fetch_vote_account_metas(bank, epoch);
+    let jito_mev_metas = fetch_jito_mev_metas(bank, epoch)?;
+
+    let mut validator_metas = vote_account_metas
+        .into_iter()
+        .map(|vote_account_meta| ValidatorMeta {
+            vote_account: vote_account_meta.vote_account,
+            commission: vote_account_meta.commission,
+            mev_commission: jito_mev_metas
+                .iter()
+                .find(|jito_mev_meta| jito_mev_meta.vote_account == vote_account_meta.vote_account)
+                .map(|jito_mev_meta| Some(jito_mev_meta.mev_commission))
+                .unwrap_or_else(|| {
+                    error!(
```

Comment:
```suggestion
                    warn!(
```
this is expected as many validators do not run jito

Reply:
Oh, I see.